calyx/src/passes/well_formed.rs

```diff
@@ -78,6 +80,37 @@ where
     Ok(())
 }
 
+fn same_binding(binding_out: &SmallVec<[(ir::Id, u64); 5]>, binding_in:&SmallVec<[(ir::Id, u64); 5]>) -> bool {
+    if (*binding_out).len() != (*binding_in).len() {
+        return false;
+    }
+
+    //println!("{}",(*binding_out).len());
+
+    for i in 0 .. (*binding_out).len() {
```

Comment:
You can use the zip method in Iterator to combine two iterators and get elements from both at the same time